scalding-core/src/main/scala/com/twitter/scalding/serialization/RequiredBinaryComparators.scala

```diff
@@ -0,0 +1,37 @@
+package com.twitter.scalding.serialization
+
+import com.twitter.scalding._
+
+import scala.language.experimental.{macros => smacros}
+
+/**
+ * RequiredBinaryComparators provide comparators (or Ordering in Scala) that are capable of comparing keys in their
+ serialized form reducing the amount of time spent in serialization/deserialization.  These comparators are implemented
+ using Scala macros, and currently provide binary comparators for primitives, strings, Options, tuples, collections, case classes
+ and Scrooge objects.
+ */
+trait RequiredBinaryComparators extends Job {
+
+  implicit def ordSer[T]: OrderedSerialization[T] = macro com.twitter.scalding.serialization.macros.impl.OrderedSerializationProviderImpl[T]
+
+  override def config =
+    super.config + (Config.ScaldingRequireOrderedSerialization -> "true")
+}
+
+object RequiredBinaryComparators {
+
+  implicit def ordSer[T]: OrderedSerialization[T] = macro com.twitter.scalding.serialization.macros.impl.OrderedSerializationProviderImpl[T]
```

Comment:
can we change the name of this to say, `orderedSerialization`? Scala has this issue (bug?) where if you have two implicits in scope of the same name you get issues:
```
scala> object Test { implicit val name: String = "object" }
defined module Test

scala> trait TestT { implicit val name: String = "trait" }
defined trait TestT

scala> class TestClass extends TestT { import Test._; println(implicitly[String]) }
<console>:9: error: could not find implicit value for parameter e: String
       class TestClass extends TestT { import Test._; println(implicitly[String]) }
                                                                        ^

scala> import Test._
import Test._

scala> class TestClass extends TestT { println(implicitly[String]) }
defined class TestClass

scala> new TestClass
trait
res0: TestClass = TestClass@20c9cd89

scala> class TestClass extends TestT { import Test._; println(implicitly[String]) }
<console>:12: error: could not find implicit value for parameter e: String
       class TestClass extends TestT { import Test._; println(implicitly[String]) }
                                                                        ^

scala> object Test2 { val name2: String = "Test2" }
defined module Test2

scala> class TestClass extends TestT { import Test2._; println(implicitly[String]) }
defined class TestClass

scala> new TestClass
trait
res1: TestClass = TestClass@3ba4236f
```